servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -502,6 +509,35 @@ public long createFile(TachyonURI path, long blockSizeBytes, boolean recursive,
   }
 
   /**
+   * Reinitializes the blocks of an existing open file.
+   *
+   * @param path the path to the file
+   * @param blockSizeBytes the new block size
+   * @param ttl the ttl
+   * @return the file id
+   * @throws InvalidPathException
+   */
+  public long reinitializeFile(TachyonURI path, long blockSizeBytes, long ttl)
+      throws InvalidPathException {
+    // TODO(yupeng): add validation
+    synchronized (mInodeTree) {
+      long id = mInodeTree.reinitializeFile(path, blockSizeBytes, ttl);
+      writeJournalEntry(new ReinitializeBlockEntry(path.getPath(), blockSizeBytes, ttl));
+      flushJournal();
+      return id;
+    }
+  }
+
+  private void resetBlockSizeFromEntry(ReinitializeBlockEntry entry) {
```

Comment:
The method name doesn't seem to describe what the method is doing? (touches more than just block size)